### ISSUE TITLE
driver.set_clipboard() and driver.set_clipboard_text doesn't work

### ISSUE THREAD

**Issue by Mywayking** [appium/python-client#265]
## The problem

Briefly describe the issue you are experiencing (or the feature you want to see added to Appium). Tell us what you were trying to do and what happened instead. Remember, this is _not_ a place to ask questions. For that, go to http://discuss.appium.io!

## Environment
- Appium version (or git revision) that exhibits the issue:
- Last Appium version that did not exhibit the issue (if applicable):
- Desktop OS/version used to run Appium:
- Node.js version (unless using Appium.app|exe):
- Mobile platform/version under test:
- Real device or emulator/simulator:
- Appium CLI or Appium.app|exe:

## Details

If necessary, describe the problem you have been experiencing in more detail.

## Link to Appium Logs

Create a [GIST](https://gist.github.com) which is a paste of your _full_ Appium logs, and link them here.

**Comment by KazuCocoa** [comment]
Can you attach the server & client logs?

**Comment by JeanMaritain** [comment]
### driver.set_clipboard_text() not successful
throws: TypeError: Object of type 'bytes' is not JSON serializable
### enviroment
- Appium Version 1.6.3
- Appium-Python-Client 0.29 
- Python 3.6
 ### python client
```call stack
  File "/Users/superuser/PycharmProjects/MobileAutoTest/pageobjects/simplepgs.py", line 68, in set_clipboard
    self.driver.set_clipboard(bytes(content, 'utf8'))
  File "/Users/superuser/PycharmProjects/MobileAutoTest/venv/lib/python3.6/site-packages/appium/webdriver/webdriver.py", line 1279, in set_clipboard
    self.execute(Command.SET_CLIPBOARD, options)
  File "/Users/superuser/PycharmProjects/MobileAutoTest/venv/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py", line 319, in execute
    response = self.command_executor.execute(driver_command, params)
  File "/Users/superuser/PycharmProjects/MobileAutoTest/venv/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py", line 374, in execute
    data = utils.dump_json(params)
  File "/Users/superuser/PycharmProjects/MobileAutoTest/venv/lib/python3.6/site-packages/selenium/webdriver/remote/utils.py", line 33, in dump_json
    return json.dumps(json_struct)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'bytes' is not JSON serializable
```

### appium server

```log

[ADB] Running '/Users/superuser/Library/Android/sdk/platform-tools/adb -P 5037 -s c7916adb shell am force-stop io.ionic.starter'
[Appium] Closing session, cause was 'UiAUtomator shut down unexpectedly'
[Appium] Removing session 01276fe8-b60e-46f0-a33f-93afae3b2afb from our master session list
[ADB] Running '/Users/superuser/Library/Android/sdk/platform-tools/adb -P 5037 -s c7916adb shell kill 21190'
[ADB] Running '/Users/superuser/Library/Android/sdk/platform-tools/adb -P 5037 -s c7916adb shell kill 21190'
[UiAutomator] Starting UIAutomator
[ADB] Creating ADB subprocess with args: ["-P",5037,"-s","c7916adb","shell","uiautomator","runtest","AppiumBootstrap.jar","-c","io.appium.android.bootstrap.Bootstrap","-e","pkg","io.ionic.starter","-e","disableAndroidWatchers",false,"-e","acceptSslCerts",false]
[ADB] Pressing the HOME button
[ADB] Running '/Users/superuser/Library/Android/sdk/platform-tools/adb -P 5037 -s c7916adb shell input keyevent 3'
[UiAutomator] Moving to state 'online'
[AndroidBootstrap] [BOOTSTRAP LOG] [debug] Loading json...
[AndroidBootstrap] [BOOTSTRAP LOG] [debug] json loading complete.
[AndroidBootstrap] [BOOTSTRAP LOG] [debug] Registered crash watchers.
[AndroidBootstrap] Android bootstrap socket is now connected
[ADB] Running '/Users/superuser/Library/Android/sdk/platform-tools/adb -P 5037 -s c7916adb shell dumpsys window'
[AndroidBootstrap] [BOOTSTRAP LOG] [debug] Client connected
[AndroidDriver] Screen already unlocked, doing nothing
[ADB] Device API level: 25
[ADB] Running '/Users/superuser/Library/Android/sdk/platform-tools/adb -P 5037 -s c7916adb shell am start -W -n io.ionic.starter/io.ionic.starter.MainActivity -S'
[AndroidBootstrap] Cannot shut down Android bootstrap; it has already shut down
[Logcat] Stopping logcat capture
[ADB] Running '/Users/superuser/Library/Android/sdk/platform-tools/adb -P 5037 -s c7916adb shell am force-stop io.appium.unlock'
[AndroidDriver] Not cleaning generated files. Add `clearSystemFiles` capability if wanted.
[Appium] New AndroidDriver session created successfully, session c3fd0be2-7809-47ab-9b8f-3b002efda584 added to master session list
[BaseDriver] Event 'newSessionStarted' logged at 1540887948363 (16:25:48 GMT+0800 (CST))
[MJSONWP] Responding to client with driver.createSession() result: {"platform":"LINUX","webStorageEnabled":false,"takesScreenshot":true,"javascriptEnabled":true,"databaseEnabled":false,"networkConnectionEnabled":true,"locationContextEnabled":false,"warnings":{},"desired":{"platformName":"Android","platformVersion":"7.1.1","deviceName":"note3","noReset":false,"appActivity":"io.ionic.starter.MainActivity","appPackage":"io.ionic.starter","app":"/Users/superuser/dev/src/Elastos.ORG.Wallet.Mobile/platforms/android/app/build/outputs/apk/debug/app-debug.apk","newCommandTimeout":6000},"platformName":"Android","platformVersion":"7.1.1","deviceName":"c7916adb","noReset":false,"appActivity":"io.ionic.starter.MainActivity","appPackage":"io.ionic.starter","app":"/Users/superuser/dev/src/Elastos.ORG.Wallet.Mobile/platforms/android/app/build/outputs/apk/debug/app-debug.apk","newCommandTimeout":6000,"deviceUDID":"c7916adb","deviceScreenSize":"1080x1920","deviceModel":"Mi Note 3","deviceManufacturer":"Xiaomi"}
[HTTP] <-- POST /wd/hub/session 200 8596 ms - 1013
[HTTP] 
```

**Comment by KazuCocoa** [comment]
I've created a PR => https://github.com/appium/python-client/pull/267

**Comment by KazuCocoa** [comment]
should be fixed in v0.30
